frame/election-provider-support/src/lib.rs

```diff
@@ -471,16 +471,15 @@ where
 	}
 }
 
-/// A utility trait for something to implement `ElectionDataProvider` in a sensible way.
-///
-/// This is generic over `AccountId` and it can represent a validator, a nominator, or any other
-/// entity.
+/// A trait that defines a set of functions that allow read access to the underlying storage
+/// and the types those functions depend upon.
```

Comment:
I think this description is waaay to generic. There is not concept of "storage" here. 

I would move the old doc to `SortedListProvider`, and just document this one as "read only subset of `SortedListProvider`".

Reply:
Each SortedList provider is backed by some sort of storage, in case of BagsList it's whatever BagsList is using to store it's nodes.
In case of UseValidatorsMap and the likes it could be Staking storage maps. 

However I get where you are coming from, we don't need to add any extra docs to this. I'll fix this by referencing SortedListProvider. The main reason I didn't before is that since we are forced to inherit SortedListProvider from ReadOnlySortedListProvider - the rustdoc refrences then are a bit backwards, as it does not understand inheritance, so if the method `iter()` is defined in ReadOnly - we have to reference `ReadOnlineSortedListProvider::iter()` as opposed to `SortedListProvider::iter()`.

Reply:
The first like documenting a trait should be a _clear, concise_ explanation of _what it is_. This trait is called `SortedListProvider`. Its job, foremost, is to provide a sorted list over some `AccountId` type. 

Honestly, the phrase 

>  A trait that defines a set of functions that allow read access to the underlying storage
> /// and the types those functions depend upon.


is irrelevant to any of this, and is just making things more confusing.

Reply:
Ping. ^^

Reply:
Outdated diff, already addressed